vnext/Microsoft.ReactNative/Fabric/Composition/SwitchComponentView.cpp

```diff
@@ -0,0 +1,249 @@
+  // to specify the entire surface, which nullptr is shorthand for (but, as it works out,
+  // any time we make an update we touch the entire surface, so we always pass nullptr).
+  winrt::com_ptr<ID2D1DeviceContext> d2dDeviceContext;
+  POINT offset;
+
+  winrt::com_ptr<Composition::ICompositionDrawingSurfaceInterop> drawingSurfaceInterop;
+  m_drawingSurface.as(drawingSurfaceInterop);
+
+  if (CheckForDeviceRemoved(drawingSurfaceInterop->BeginDraw(d2dDeviceContext.put(), &offset))) {
+    const auto switchProps = std::static_pointer_cast<const facebook::react::SwitchProps>(m_props);
+
+    d2dDeviceContext->Clear(D2D1::ColorF(D2D1::ColorF::Black, 0.0f));
+    if (m_props->backgroundColor) {
+      d2dDeviceContext->Clear(m_props->backgroundColor.AsD2DColor());
+    }
+
+    float offsetX = static_cast<float>(offset.x / m_layoutMetrics.pointScaleFactor);
+    float offsetY = static_cast<float>(offset.y / m_layoutMetrics.pointScaleFactor);
+
+    // https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/CommonStyles/ToggleSwitch_themeresources.xaml
+    constexpr float thumbMargin = 3.0f;
+    constexpr float thumbRadius = 7.0f;
+    constexpr float trackWidth = 40.0f;
+    constexpr float trackHeight = 20.0f;
+    constexpr float trackCornerRadius = 10.0f;
+
+    auto frame{m_layoutMetrics.frame.size};
+    float trackMarginX = (frame.width - trackWidth) / 2;
+    float trackMarginY = (frame.height - trackHeight) / 2;
+
+    D2D1_RECT_F trackRect = D2D1::RectF(
+        offsetX + trackMarginX,
+        offsetY + trackMarginY,
+        offsetX + trackMarginX + trackWidth,
+        offsetY + trackMarginY + trackHeight);
+
+    // switchProps->value = false
+    float thumbX = trackRect.left + thumbMargin + thumbRadius;
+
+    if (switchProps->value) {
+      thumbX = trackRect.right - thumbMargin - thumbRadius;
+    }
+
+    winrt::com_ptr<ID2D1SolidColorBrush> defaultBrush;
+
+    D2D1_COLOR_F defaultColor =
+        switchProps->disabled ? facebook::react::greyColor().AsD2DColor() : facebook::react::blackColor().AsD2DColor();
+
+    winrt::check_hresult(d2dDeviceContext->CreateSolidColorBrush(defaultColor, defaultBrush.put()));
+
+    winrt::com_ptr<ID2D1SolidColorBrush> thumbBrush;
+    if (!switchProps->disabled && switchProps->thumbTintColor) {
+      winrt::check_hresult(
+          d2dDeviceContext->CreateSolidColorBrush(switchProps->thumbTintColor.AsD2DColor(), thumbBrush.put()));
+    } else {
+      thumbBrush = defaultBrush;
+    }
+
+    const auto dpi = m_layoutMetrics.pointScaleFactor * 96.0f;
+    float oldDpiX, oldDpiY;
+    d2dDeviceContext->GetDpi(&oldDpiX, &oldDpiY);
+    d2dDeviceContext->SetDpi(dpi, dpi);
+
+    // switch thumb
+    D2D1_POINT_2F thumbCenter = D2D1 ::Point2F(thumbX, (trackRect.top + trackRect.bottom) / 2);
+    D2D1_ELLIPSE thumb = D2D1::Ellipse(thumbCenter, thumbRadius, thumbRadius);
+    d2dDeviceContext->FillEllipse(thumb, thumbBrush.get());
```

Comment:
I think I'd have implemented the thumb as a child visual, that way it could be animated when it switches value?

Reply:
Created #11210 to address this in a separate PR